### PR TITLE
Issue #12 Wrong counting in Samples Indices graph

### PR DESCRIPTION
**Pull request by ijafri26**
Issue #12 
Grouped data for Samples Indices graph (line 194) by both index and library to prevent incorrect counting of data.

### CODE REVIEW

File: bcl2fastq/index_summary.py [hunk 1]
```diff
@@ -189,13 +189,15 @@ def update_known_index_bar(run_json):
...
 
     data = []
-    for inx, d in run.groupby('index'):
+    #   Multiple libraries can use the same index, data must be grouped by both index and libraries
+    #   to prevent duplicate counts
+    for inx, d in run.groupby(['index', 'library']):
         data.append({
             'x': list(d['library']),
             # One library can be run on multiple lanes. Sum them together.
             'y': list(d.groupby('library')['SampleNumberReads'].sum()),
```
Comment (slazicoicr) [inline bcl2fastq/index_summary.py]: Since you already grouped by library, there is no need to group again.

**Comment by callunity** [comment]
> Use the [re-request review function](https://help.github.com/en/articles/requesting-a-pull-request-review) (Point # 7) to indicate that your commits are ready for another review.

@slazicoicr am I missing something? It doesn't look like the commits had been reviewed a first time, or that you had been previous requested for a review, so I'm not clear on why she'd need to request a re-review...

**Comment by slazicoicr** [comment]
My understanding of this feature was that since I asked for changes in this pull request, Irene could commit several times. I would not know when she is ready for this Pull Request to be reviewed again. By explicitly pressing the re-review button on my name, she would be stating that I should review the Pull Request again.

**Comment by callunity** [comment]
Ah, I see; I thought you were commenting on her previous use of the review request feature, as opposed to requesting a specific use pattern going forward. This makes more sense.

**Comment by slazicoicr** [comment]
I think this approach fixes the bug. @ijafri26 let us know how this differs from your approach that failed to produce valid outputs:

```
     data = []
-    for inx, d in run.groupby('index'):
+    for inx, d in run.groupby(['index', 'library']):
         data.append({
-            'x': list(d['library']),
+            'x': list(d['library'].unique()),
             # One library can be run on multiple lanes. Sum them together.
-            'y': list(d.groupby('library')['SampleNumberReads'].sum()),
+            'y': [d['SampleNumberReads'].sum()],
             'type': 'bar',
-            'name': inx,
+            'name': inx[0],
             'marker': {'line': {
                 'width': 2,
                 'color': 'rgb(255,255, 255)'
```

**Pull request merged**